### PR TITLE
TF-3008 Search to chip

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3008 

## Resolved

- Web:

https://github.com/user-attachments/assets/f309d377-7b31-409c-aa90-e94b53c5380c

- Mobile:

[demo-android.webm](https://github.com/user-attachments/assets/39fff3e7-6a65-4319-bb16-d5e929c0cb4f)

### CODE REVIEW

File: lib/features/mailbox_dashboard/presentation/widgets/search_input_form_widget.dart [hunk 1]
```diff
@@ -135,13 +136,23 @@ class SearchInputFormWidget extends StatelessWidget with AppLoaderMixin {
...
+      } else {
+        _dashBoardController.searchEmail(queryString: query);
+      }
     } else {
       _dashBoardController.clearSearchEmail();
     }
   }
 
+
+
```
Comment (tddang-linagora) [inline lib/features/mailbox_dashboard/presentation/widgets/search_input_form_widget.dart:155]: Delete whitespace

File: lib/features/mailbox_dashboard/presentation/widgets/search_input_form_widget.dart [hunk 2]
```diff
@@ -151,12 +162,19 @@ class SearchInputFormWidget extends StatelessWidget with AppLoaderMixin {
...
     _searchController.applyFilterSuggestionToSearchFilter(_dashBoardController.sessionCurrent?.username);
-    _dashBoardController.searchEmail(queryString: recent.value);
+
+    if (EmailUtils.isEmailAddressValid(recent.value)) {
+      _searchController.searchInputController.clear();
+      _dashBoardController.searchEmail(emailAddress: EmailAddress(null, recent.value));
+    } else {
+      _searchController.searchInputController.text = recent.value;
+      _dashBoardController.searchEmail(queryString: recent.value);
+    }
```
Comment (tddang-linagora) [inline lib/features/mailbox_dashboard/presentation/widgets/search_input_form_widget.dart:177]: Code duplication. Should refactor.

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 3]
```diff
@@ -665,15 +665,29 @@ class MailboxDashBoardController extends ReloadableController with UserSettingPo
...
+      searchController.updateFilterEmail(
+        sortOrderOption: searchController.sortOrderFiltered.value.getSortOrder()
+      );
+    }
+    if (emailAddress?.emailAddress.isNotEmpty == true) {
+      searchController.updateFilterEmail(toOption: Some({emailAddress!.emailAddress}));
+      dispatchAction(StartSearchEmailAction(filter: QuickSearchFilter.to));
+    } else {
+      dispatchAction(StartSearchEmailAction());
     }
```
Comment (tddang-linagora) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:690]: If both `queryString` and `emailAddress` are inputted, these if checks (`queryString?.isNotEmpty == true` and  `emailAddress?.emailAddress.isNotEmpty == true`) are both executed. Is this expected behavior? If not, consider using `else if` or `assert` statement.

File: lib/features/search/email/presentation/search_email_controller.dart [hunk 4]
```diff
@@ -463,22 +463,50 @@ class SearchEmailController extends BaseController
...
+    } else {
+      setTextInputSearchForm(recentSearch.value);
+      _updateSimpleSearchFilter(
+        textOption: Some(SearchQuery(recentSearch.value)),
+        beforeOption: const None(),
+        positionOption: emailSortOrderType.value.isScrollByPosition()
+          ? const Some(0)
+          : const None()
+      );
+    }
```
Comment (tddang-linagora) [inline lib/features/search/email/presentation/search_email_controller.dart:509]: Code duplication, consider refactor.

File: lib/features/mailbox_dashboard/presentation/widgets/search_input_form_widget.dart (hunk 2 above)
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/widgets/search_input_form_widget.dart]: Good catch. Done

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart (hunk 3 above)
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart]: Good catch. Done

File: lib/features/search/email/presentation/search_email_controller.dart (hunk 4 above)
Comment (dab246) [reply lib/features/search/email/presentation/search_email_controller.dart]: Done

**Review by tddang-linagora — APPROVED** [review]

**Comment by dab246** [comment]
This PR resolve issue #3006. Issue #3008 is resolve together with #3002. So we will close this PR and create a new PR for issue #3006

**Pull request closed without merging**